dask/bag/core.py

```diff
@@ -751,6 +758,90 @@ def write(data, filename):
         f.close()
 
 
+def from_s3(bucket_name, paths='*', aws_access_key=None, aws_secret_key=None,
+            connection=None, anon=False):
+    """ Create a dask.bag by loading files from s3.
+
+    >>> b = from_s3(bucket, 'myfile1.txt')
+
+    or a pattern with '*' or '?'. There is one partition per file.
+
+    >>> b = from_s3(bucket, '*.json')
+    >>> b.npartitions
+    9
+
+    or a list of files.
+
+    >>> b = from_s3(bucket, ['alice.csv', 'bob.csv'])
+    >>> b.npartitions
+    2
+
+    """
+    def _connect_s3(aws_access_key, aws_secret_key, connection, anon):
+        """Connect to s3"""
+        import boto
+        if anon is True:
+            connection = boto.connect_s3(anon=anon)
+        elif connection is None:
+            connection = boto.connect_s3(aws_access_key, aws_secret_key)
+        return connection
+
+    def _from_s3(bucket_name, matches, conn_args):
+        """
+        takes an s3 bucket object and a list of bucket strings
+        """
+        dsk = {}
+        for i, m in enumerate(matches):
+            # connect to s3
+            a, b, c, d = conn_args
+            get_connection = (_connect_s3, a, b, c, d)
+            # get the bucket with the connection
+            get_bucket = (lambda x: x.get_bucket(bucket_name), get_connection)
+            # get a key from the bucket
+            get_key = (lambda b, k: b.get_key(k), get_bucket, m)
+            # read the key
+            read_key = lambda k: [getattr(k, 'read')()]
```

Comment:
I think what I would want to `memoize` would be a function that connects and gets a bucket. Because we get one bucket per connection, and we get it everytime we make a connection. 

But when I 
```python
@toolz.memoize
def _get_s3_bucket(...):
```
I get serialization errors. So I don't think it `memoize` is working per process...?